alert-combiner-py/src/utils.py

```diff
@@ -0,0 +1,88 @@
+from web3 import Web3
+from hexbytes import HexBytes
+import requests
+import logging
+import json
+
+etherscan_label_api = "https://api.forta.network/labels/state?sourceIds=etherscan,0x6f022d4a65f397dffd059e269e1c2b5004d822f905674dbf518d968f744c2ede&entities="
+
+class Utils:
+    CONTRACT_CACHE = dict()
+    TOTAL_SHARDS = None
+
+    @staticmethod
+    def is_contract(w3, addresses) -> bool:
+        """
+        this function determines whether address/ addresses is a contract; if all are contracts, returns true; otherwise false
+        :return: is_contract: bool
+        """
+        if addresses is None:
+            return True
+
+        if Utils.CONTRACT_CACHE.get(addresses) is not None:
+            return Utils.CONTRACT_CACHE[addresses]
+        else:
+            is_contract = True
+            for address in addresses.split(','):
+                code = w3.eth.get_code(Web3.toChecksumAddress(address))
+                is_contract = is_contract & (code != HexBytes('0x'))
+            Utils.CONTRACT_CACHE[addresses] = is_contract
+            return is_contract
+        
+    @staticmethod
+    def is_address(w3, addresses: str) -> bool:
+        """
+        this function determines whether address is a valid address
+        :return: is_address: bool
+        """
+        if addresses is None:
+            return True
+
+        is_address = True
+        for address in addresses.split(','):
+            for c in ['a', 'b', 'c', 'd', 'e', 'f', '0', '1', '2', '3', '4', '5', '6', '7', '8', '9']:
```

Comment:
this is my super ugly code; do you know how to better check this?

Reply:
I think you can go for:

```python
_test = any([9*c in address.lower() for c in 'abcdef0123456789'])
if _test: pass
```

I fell for the trap of the regex too, it's 8 repetitions! :sweat_smile: 

```python
_r = re.compile(r'([a-f0-9])\1{8}')
```